src/pyramid/static.py

```diff
@@ -94,46 +143,128 @@ def __call__(self, context, request):
         if path is None:
             raise HTTPNotFound('Out of bounds: %s' % request.url)
 
+        # normalize asset spec or fs path into resource_path
         if self.package_name:  # package resource
             resource_path = '%s/%s' % (self.docroot.rstrip('/'), path)
             if resource_isdir(self.package_name, resource_path):
                 if not request.path_url.endswith('/'):
-                    self.add_slash_redirect(request)
+                    raise self.add_slash_redirect(request)
```

Comment:
This changes the behaviour from previous versions of Pyramid, will this break anything?

Reply:
It's not different, I just moved the raise outside the method instead of inside to be more explicit.

Reply:
Oh, I see now. I missed it. :+1: